src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -1,18 +1,246 @@
+    pub fn lease_id(&self) -> i64 {
+        self.inner.lease_id
+    }
+
+    /// Get the ttl
+    #[must_use]
+    pub fn ttl(&self) -> u64 {
+        self.inner.ttl
+    }
+
+    /// Get the `is_closed` flag
+    #[must_use]
+    pub fn is_closed(&self) -> bool {
+        self.inner.is_closed()
+    }
+
+    /// Try to close session
+    pub fn close(&self) {
+        match self.close_tx.try_send(()) {
+            Ok(o) => {
+                info!("close the keep alive session, {o:?}");
+            }
+            Err(e) => {
+                error!("failed to close the keep alive session, error={e:?}");
+            }
+        }
+        self.inner.close();
+    }
+}
+
+impl Drop for Session {
+    fn drop(&mut self) {
+        // Set the close flag
+        self.close();
+    }
+}
+
+/// The keepalive session inner struct
+pub struct SessionInner {
+    /// The closed flag
+    closed: Arc<AtomicBool>,
+    /// Current lease id
+    lease_id: i64,
+    /// The ttl of the lease
+    ttl: u64,
+    /// The etcd client
+    #[cfg_attr(not(debug_assertions), skip)]
+    client: etcd_client::Client,
+}
+
+impl Debug for SessionInner {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SessionInner")
+            .field("closed", &self.closed)
+            .field("lease_id", &self.lease_id)
+            .field("ttl", &self.ttl)
+            .finish_non_exhaustive()
+    }
+}
+
+impl SessionInner {
+    /// Get the closed flag
+    fn is_closed(&self) -> bool {
+        self.closed.load(Ordering::Acquire)
+    }
+
+    /// Set close flag
+    fn close(&self) {
+        self.closed.store(true, Ordering::Release);
+    }
+
+    /// Reopen close flag
+    fn reopen(&self) {
+        self.closed.store(false, Ordering::Release);
+    }
+
+    /// Keep alive the lease
+    #[allow(clippy::pattern_type_mismatch)] // Raised by `tokio::select`
+    pub async fn keepalive(&self, mut close_rx: mpsc::Receiver<()>, token: CancellationToken) {
+        let mut interval = tokio::time::interval(Duration::from_secs(self.ttl.div_ceil(3)));
+
+        if self.is_closed() {
+            error!("lease keep alive stream closed by is_closed flag");
+            return;
+        }
+
+        // Try to clone a client, if the keeper failed, try to reconnect
+        let mut client = self.client.clone();
+        let lease_id = self.lease_id;
+        let (mut keeper, mut lease_keep_alive_stream) =
+            match client.lease_keep_alive(lease_id).await {
+                Ok((keeper, lease_keep_alive_stream)) => (keeper, lease_keep_alive_stream),
+                Err(e) => {
+                    error!("failed to keep alive lease, error={e:?}");
+
+                    return;
+                }
+            };
+
+        // Set the is_closed flag
+        self.reopen();
+
+        // Used to revoke client
+        let mut client = self.client.clone();
+        // Start to keep alive the lease
+        loop {
```

Comment:
Add a label for break to make it more clear

Reply:
Ok, I have removed this line directly.

Reply:
I mean  use `break 'Label ` will be more clear because the loop is large